pkg/controllermanager/controller/namespacedcloudprofile/reconciler_test.go

```diff
@@ -908,6 +921,281 @@ var _ = Describe("NamespacedCloudProfile Reconciler", func() {
+
+						version := namespacedCloudProfile.Status.CloudProfileSpec.MachineImages[0].Versions[0]
+						Expect(version.CapabilityFlavors).To(HaveLen(1))
+						Expect(version.CapabilityFlavors[0].Capabilities).To(HaveKeyWithValue("architecture", gardencorev1beta1.CapabilityValues{"arm64"}))
+						Expect(version.CapabilityFlavors[0].Capabilities).To(HaveKeyWithValue("gpu", gardencorev1beta1.CapabilityValues{"nvidia"}))
+					})
+
+					It("should set architecture capabilities for machine types", func() {
+						namespacedCloudProfile.Spec.MachineTypes = []gardencorev1beta1.MachineType{
+							{
+								Name:   "m5.large",
+								CPU:    resource.MustParse("2"),
+								Memory: resource.MustParse("8Gi"),
+							},
+							{
+								Name:         "m5.arm.large",
+								CPU:          resource.MustParse("2"),
+								Memory:       resource.MustParse("8Gi"),
+								Architecture: ptr.To("arm64"),
+							},
+						}
+
+						namespacedcloudprofilecontroller.MergeCloudProfiles(namespacedCloudProfile, cloudProfile)
+
+						machineTypes := namespacedCloudProfile.Status.CloudProfileSpec.MachineTypes
+						Expect(machineTypes).To(HaveLen(2))
+
+						Expect(machineTypes).To(ConsistOf(
+							MatchFields(IgnoreExtras, Fields{
+								"Architecture": Equal(ptr.To("amd64")),
+								"Capabilities": HaveKeyWithValue("architecture", gardencorev1beta1.CapabilityValues{"amd64"}),
+							}),
+							MatchFields(IgnoreExtras, Fields{
+								"Architecture": Equal(ptr.To("arm64")),
+								"Capabilities": HaveKeyWithValue("architecture", gardencorev1beta1.CapabilityValues{"arm64"}),
+							}),
+						))
+					})
+				})
+
+				When("parent CloudProfile has no capability definitions", func() {
+					BeforeEach(func() {
+						cloudProfile.Spec.MachineCapabilities = nil
+					})
+
+					It("should extract architectures from capability flavors", func() {
+						namespacedCloudProfile.Spec.MachineImages = []gardencorev1beta1.MachineImage{
+							{
+								Name: "ubuntu",
+								Versions: []gardencorev1beta1.MachineImageVersion{
+									{
+										ExpirableVersion: gardencorev1beta1.ExpirableVersion{Version: "20.04"},
+										CapabilityFlavors: []gardencorev1beta1.MachineImageFlavor{
+											{
+												Capabilities: gardencorev1beta1.Capabilities{
+													"architecture": []string{"amd64"},
+												},
+											},
+											{
+												Capabilities: gardencorev1beta1.Capabilities{
+													"architecture": []string{"arm64"},
+												},
+											},
+										},
+									},
+								},
+							},
+						}
+
+						namespacedcloudprofilecontroller.MergeCloudProfiles(namespacedCloudProfile, cloudProfile)
+
+						version := namespacedCloudProfile.Status.CloudProfileSpec.MachineImages[0].Versions[0]
+						Expect(version.Architectures).To(ConsistOf("amd64", "arm64"))
+						Expect(version.CapabilityFlavors).To(BeNil())
+					})
```

Comment:
This is the same case as `should handle multiple capability flavors with different architectures`, hence I would suggest to remove this one.

```suggestion
```